src/main/java/org/acme/consumer/RepoMetaAnalysisStreamWork.java

```diff
@@ -0,0 +1,293 @@
+package org.acme.consumer;
+
+import io.quarkus.kafka.client.serialization.ObjectMapperSerde;
+import io.vavr.collection.Stream;
+import org.acme.model.Component;
+import org.acme.model.Repository;
+import org.acme.persistence.RepoEntityRepository;
+import org.acme.repositories.*;
+import org.acme.util.SecretsUtil;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+
+@ApplicationScoped
+public class RepoMetaAnalysisStreamWork {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RepoMetaAnalysisStreamWork.class);
+    private final RepoEntityRepository repoEntityRepository;
+
+    public RepoMetaAnalysisStreamWork(RepoEntityRepository repoEntityRepository){
+        this.repoEntityRepository = repoEntityRepository;
+    }
+
+
+    public void createStructure(StreamsBuilder streamsBuilder, MavenMetaAnalyzer mavenMetaAnalyzer,
```

Comment:
As we are referring repoEntityRepository from inside these functions, I believe we should not make it static?

Reply:
`repoEntityRepository` is the only dependency right now that is not injected as argument to this method. All other dependencies are. Instead of injecting `repoEntityRepository` via the constructor, it could be added as additional method parameter here.

Or, alternatively, inject everything via the constructor. At the moment we're mixing those two things, which is not great.